armi/reactor/blocks.py

```diff
@@ -612,6 +612,11 @@ def updateNumberDensities(self, numberDensities):
         for nucName, ndens in numberDensities.items():
             self.setNDensParam(nucName, ndens)
 
+    def updateAllNumberDensityParams(self):
+        """Update all block-level ndens params for visualization."""
```

Comment:
for visualization? I think this warrants a little more explanation. Under what circumstances (e.g. in `applyFuelDisplacement` above) should one have to call this? What happens if they don't?

Reply:
if they don't, then we'll will be even more motivated to update the DB viewing tools to use actual state data for number densities rather than these maybe-correct copies.

Reply:
I think that `buildNumberDensityParams` already exists on `blocks` for this purpose. Can it be reused instead?

Reply:
aha I knew that was in there somewhere. thanks!!